parser-typechecker/src/Unison/Codebase/Branch.hs

```diff
@@ -501,8 +510,16 @@ empty0 =
   Branch0 mempty mempty mempty mempty mempty mempty mempty mempty mempty mempty
 
 -- | Checks whether a Branch0 is empty.
+-- This is not as easy as checking whether the branch is equal to the `empty0` branch
+-- because child branches may be empty, but still have history.
 isEmpty0 :: Branch0 m -> Bool
-isEmpty0 = (== empty0)
+isEmpty0 (Branch0 _terms _types _children _edits deepTerms deepTypes deepTermMetadata deepTypeMetadata deepPaths deepEdits) =
```

Comment:
It's worth taking a look at the changed logic here and within the different `derive` combinators.